src/copilot/app/indexing_api.py

```diff
@@ -76,6 +85,136 @@ async def init_indexing():
+    
+    embedding_column = "embedding" in data.fieldnames
+    language_column = "language" in data.fieldnames
+    tag_column = "tag" in data.fieldnames
+
+    for row in data:
+        embedding = ast.literal_eval(row["embedding"]) if embedding_column else None
+        language = row["language"] if language_column else None
+        tag = row["tag"] if tag_column else None
+        
+        document = DocumentCreate(url=row["url"], text=row["text"], embedding=embedding, source=file.filename, language=language, tag=tag)
+        document_service.upsert(db, document, embed=embed)
+
+    file.file.close()
+    return {"content": "yay"}
+
+
+@app.post("/upload_csv_faq", summary="Upload a CSV file for FAQ data", status_code=200, response_model=ResponseBody)
+def upload_csv_faq(file: UploadFile = File(...), embed: bool = False, db: Session = Depends(get_db)):
+    """
+    Upload a CSV file containing RAG data to the database.
+
+    Parameters
+    ----------
+    file : UploadFile
+        The CSV file sent by the user
+    embed : bool, optional
+        Whether to embed the data or not. Defaults to False.
+    db : Session
+        Database session
+
+    Returns
+    -------
+    ResponseBody
+        A response body containing a confirmation message upon successful completion of the process.
+    """
+    data = csv.DictReader(codecs.iterdecode(file.file, 'utf-8'))
+    
+    embedding_column = "embedding" in data.fieldnames
+    language_column = "language" in data.fieldnames
+    tag_column = "tag" in data.fieldnames
+
+    for row in data:
+        embedding = ast.literal_eval(row["embedding"]) if embedding_column else None
+        language = row["language"] if language_column else None
+        tag = row["tag"] if tag_column else None
+        
+        question = QuestionCreate(url=row["url"], text=row["text"], answer=row["answer"], embedding=embedding, source=file.filename, language=language, tag=tag)
+        question_service.upsert(db, question, embed=embed)
+
+    file.file.close()
+    return {"content": "yay"}
+
+
+@app.post("/upload_pdf_rag", summary="Upload a PDF file for RAG data", status_code=200, response_model=ResponseBody)
+def upload_pdf_rag(file: UploadFile = File(...), embed: bool = False, db: Session = Depends(get_db)):
+    """
+    Upload a CSV file containing RAG data to the database.
+
+    Parameters
+    ----------
+    file : UploadFile
+        The CSV file sent by the user
+    embed : bool, optional
+        Whether to embed the data or not. Defaults to False.
+    db : Session
+        Database session
+
+    Returns
+    -------
+    ResponseBody
+        A response body containing a confirmation message upon successful completion of the process.
+    """
+    with tempfile.NamedTemporaryFile(delete=False, suffix=".pdf") as temp_file:
+        temp_filename = temp_file.name
+        shutil.copyfileobj(file.file, temp_file)
+
+    documents = PyPDFToDocument().run(sources=[Path(temp_filename )])
```

Comment:
Why not use the indexer object as specified in the config? So that RAG document preprocessing is consistent.